restful-order-service/backend_server/server.bal

```diff
@@ -0,0 +1,58 @@
+import ballerina/http;
+
+type Order record {|
+    readonly string id;
+    string customerId;
+    string? shipId;
+    string date;
+    OrderStatus status;
+    int quantity;
+    string item;
+|};
+
+enum OrderStatus {
+    PENDING,
+    SHIPPED,
+    DELIVERED,
+    CANCELED,
+    RETURNED
+};
+
+final table<Order> key(id) orders = table [
+    {id: "HM-278", quantity: 5, item: "TV", customerId: "C-124", shipId: "S-8", date: "22-11-2023", status: PENDING},
+    {id: "HM-340", quantity: 3, item: "IPhone 14", customerId: "C-73", shipId: "S-32", date: "12-11-2023", status: DELIVERED}
+];
+
+@http:ServiceConfig {
+    cors: {
+        allowOrigins: ["*"]
+    }
+}
+service /sales on new http:Listener(9090) {
+
+    // Example: http://localhost:9090/sales/orders
+    resource function get orders() returns Order[] {
+        return orders.toArray();
+    };
+
+    // Example: http://localhost:9090/sales/orders/HM-238
+    resource function get orders/[string id]() returns Order|http:NotFound {
+        return orders[id] ?: <http:NotFound>{body: string `Order not found. Order ID: ${id}`};
+    };
+
+    // Example: http://localhost:9090/sales/customers/C-124/orders?status=PENDING
```

Comment:
Since PENDING is the default, may be better to use a different example?

Reply:
Updated